static/app/stores/pageFiltersStore.tsx

```diff
@@ -72,20 +79,27 @@ interface PageFiltersStoreDefinition extends StrictStoreDefinition<PageFiltersSt
     pinned: Set<PinnedPageFilter>,
     persist?: boolean
   ): void;
+  onInitializeUrlStateWithCodecovData(
```

Comment:
It feels wrong to me that we separate codecov and sentry selections in these functions, yet the underlying state is a combination of the two. But the only other option I can think of is creating a completely separate store to track the codecov selection which is probably prohibitively difficult. 

Not requesting changes based on this, just noting my thoughts about it. Tagging the frontend github team to see if anyone else has good ideas.

Reply:
Yeah it's not the most elegant solution but it's the one that caused less friction.

I tried adding an extra param to the existing initialize function, but that affected a bunch of other tests - it's an option but I'm not the biggest fan.

Alternatively, I had fused the newSelection param to expect type SentryFilters + CodecovFilters, and inside the initialize fn create a helper that split that object into 'sentrySelection' and 'codecovSelection', but I wasn't too convinced with this one either.

Reply:
I agree here with @malwilley, this feels wrong.

>I tried adding an extra param to the existing initialize function, but that affected a bunch of other tests - it's an option but I'm not the biggest fan. Luckily this function is only used once, so it'd only affect one code file there.

Understood, but that is somewhat expected, as you are touching some code that is used very often at Sentry. Updating a ton of tests isn't great, and I get that it's tedious, but it's a lot easier than maintaining two different implementations under the hood.